enhancements/proposals/1277-expose-scyllacluster/README.md

```diff
@@ -0,0 +1,398 @@
+
+### Risks and Mitigations
+
+#### ScyllaDB version requirement
+
+Node replacement procedure alignment requires the newest 5.2 ScyllaDB Open Source and 2023.1 Enterprise, meaning node
+replacement procedure won’t work when Scylla doesn’t meet minimum version criteria. Users wanting to update to a newer
+Operator version will need to update their ScyllaDB versions. We will mention it in release notes.
+
+## Design Details
+
+### Network connectivity cases
+
+#### Private Kubernetes
+
+This is the current setup type (Scylla Operator <= 1.10), only Pods deployed in the same K8s cluster can connect to
+Scylla Pods. MultiDC is not possible, as Service ClusterIP are virtual IPs which are not routable outside the k8s
+cluster.
+
+![Private-K8s](private-k8s.svg "Private K8s")
+
+##### ScyllaDB configuration
+
+```yaml
+broadcast_address: ClusterIP
+broadcast_rpc_address: ClusterIP
+```
+
+---
+
+#### Private VPC
+
+In Private VPC type, Scylla Pods broadcast their ephemeral PodIP both to clients and other nodes. Because PodIPs are
+natively routable within a VPC, MultiDC is possible over private IPs. Public Internet access is not possible.
+This assumes Scylla supports UUID Node Identifiers and ephemeral IPs.
+Scylla Operator is configured to create Headless Services.
+
+![Private-VPC](private-vpc.svg "Private VPC")
+
+##### ScyllaDB configuration
+
+```yaml
+broadcast_address: PodIP
+broadcast_rpc_address: PodIP
+```
+
+---
+
+#### Private VPC + Public LB
+
+Scylla Nodes talk to each other over private IP, but all the clients, no matter where they are deployed, need to talk
+through the public Internet. MultiDC is possible over private IPs.
+Big disadvantage of this type is the cost associated with the necessity of going through the public Internet, and the
+cost of Load Balancer Services required for each Node in the ScyllaCluster.
+Scylla Operator is configured to create LoadBalancer Services.
+
+![Private-VPC+Public-LB](private-vpc-public-lb.svg "Private VPC + Public LB")
+
+##### ScyllaDB configuration
+
+```yaml
+broadcast_address: PodIP
+broadcast_rpc_address: LoadBalancer External IP
+```
+
+---
+
+#### Private VPC + Public SNI Proxy
+
+This type allows for all client connection origins, can establish MultiDC connection using private IPs, while being cost-efficient. 
+VMs deployed in the same VPC can either go directly to the pods, or use SNI proxy with a specialized driver.
+Customers either coming from public Internet or Private Link, go through SNI proxy.
+Scylla Operator is configured to create Headless Services.
+
+![Private-VPC+Public-SNI-Proxy](private-vpc-public-sni-proxy.svg "Private VPC + Public SNI Proxy")
+
+##### ScyllaDB configuration
+
+```yaml
+broadcast_address: PodIP
+broadcast_rpc_address: PodIP
+```
+
+---
+
+#### Multicloud
+
+In multicloud type we rely on any solution providing inter-VPC connectivity. Two clouds need to have distinct VPC
```

Comment:
A bit of nitpicking again, but I feel like describing the example infrastructure required for such case shouldn't be a part of design details of the proposal, or should be clearly separated. After all we don't really care how someone achieves the connectivity, but only that we set up the services and config right.

Reply:
Repharsed.